superset/views/utils.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from collections import defaultdict
+
+from flask import g
+from flask_appbuilder.security.sqla import models as ab_models
+
+from superset import db
+
+
+def bootstrap_user_data(given_username=None):
+    if given_username:
+        username = given_username
+    else:
+        username = g.user.username
+
+    user = (
+        db.session.query(ab_models.User)
+        .filter_by(username=username)
+        .one()
+    )
+    roles = {}
+    permissions = defaultdict(set)
+    for role in user.roles:
```

Comment:
For the welcome page you don't need all that in the payload, maybe this utility function should have a `include_perms=False` arg. Man I wish we had GraphQL instead of having to change the backend logic...

Reply:
💯 on the graphQL